File: 백준/Silver/1946. 신입 사원/신입 사원.py
from sys import stdin as s

T = int(s.readline())

for i in range(T):
    N = int(s.readline())
    
    arr = [list(map(int,s.readline().split())) for _ in range(N)]
    arr.sort(key=lambda x:x[0])
    cnt =1 
    min = arr[0][1]
    
    for i in range(1,N):
        if arr[i][1] <min:
            min = arr[i][1]
            cnt+=1
    
    print(cnt)

File: 백준/Silver/1946. 신입 사원/신입 사원.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int T = Integer.parseInt(br.readLine());

		for (int i = 0; i < T; i++) {
			int N = Integer.parseInt(br.readLine());
			List<int[]> arr = new ArrayList<>();

			for (int j = 0; j < N; j++) {
				String[] rank = br.readLine().split(" ");
				int docRank = Integer.parseInt(rank[0]);
				int intRank = Integer.parseInt(rank[1]);
				arr.add(new int[] {docRank, intRank});
			}

			arr.sort((a, b) -> Integer.compare(a[0], b[0]));

			int count = 0;
			int minIntRank = Integer.MAX_VALUE;
			for (int[] d : arr) {
				if (minIntRank > d[1]) {
					minIntRank = d[1];
					count++;
				}
			}

			System.out.println(count);
		}
	}
}
